src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,134 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int INITIAL_CAPACITY = 10;
+
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        this.objects = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() == objects.length) {
+            grow();
+        }
+        objects[size()] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size() || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid!");
+        }
+        if (index == size() - 1 && index == size()) {
+            this.add(value);
+        } else {
+            shift(index);
+            objects[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (!list.isEmpty()) {
+            for (int i = 0; i < list.size(); i++) {
+                this.add(list.get(i));
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (checkIndex(index)) {
+            return (T) objects[index];
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index is invalid!");
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (checkIndex(index)) {
+            objects[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid!");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (checkIndex(index)) {
+            Object element = objects[index];
+            collapse(index);
+            size--;
+            return (T) element;
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index is invalid!");
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size(); i++) {
+            if (element != null) {
+                if (element.equals(objects[i])) {
+                    return remove(i);
+                }
+            } else if (objects[i] == null) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("No Such Element!");
```

Comment:
```suggestion
        for (int i = 0; i < size; i++) {
            if (element != objects[i] || element != null && element.equals(objects[i])) {
                return remove(i);
            } 
        }
        throw new NoSuchElementException("No Such Element!");
```
lets combine those 2 ifs

Reply:
I got 3 test failed

removeObjectValueByValue()

java.lang.AssertionError: Test failed! Returned value should be Cat{name='Fantic', color='grey'} 
Expected :Cat{name='Barsik', color='black'}
Actual   :Cat{name='Fantic', color='grey'}


removeElementFromArrayListByValue()
org.junit.ComparisonFailure: Test failed! Returned value should be "Java", but was "String"
Expected :Java
Actual   :String

removeElementFromArrayListByNonExistentValue()
java.lang.Exception: Unexpected exception, expected<java.util.NoSuchElementException> but was<java.lang.AssertionError>